test/nodes/test_prompt_node.py

```diff
@@ -361,6 +361,56 @@ def test_stop_words(prompt_model):
     assert "capital" in r[0] or "Germany" in r[0]
 
 
+@pytest.mark.integration
+@pytest.mark.parametrize("prompt_model", ["openai", "azure"], indirect=True)
+def test_streaming_prompt_node_with_params(prompt_model):
+    skip_test_for_invalid_key(prompt_model)
+    stream_handler_invoked = False
+
+    def test_stream_handler(token: str) -> str:
+        # no use of print() here as we are running tests
+        # print(token, flush=True, end="")
+        nonlocal stream_handler_invoked
+        stream_handler_invoked = True
+        return token
+
+    node = PromptNode(prompt_model)
+    response = node(
+        "What are some of the best cities in the world to live and why?",
+        stream=True,
+        stream_handler=test_stream_handler,
+    )
+
+    assert len(response[0]) > 0, "Response should not be empty"
+    assert stream_handler_invoked, "Stream handler should have been invoked"
+
+
+@pytest.mark.integration
+@pytest.mark.skipif(
+    not os.environ.get("OPENAI_API_KEY", None),
+    reason="No OpenAI API key provided. Please export an env var called OPENAI_API_KEY containing the OpenAI API key.",
+)
+def test_streaming_prompt_node():
+    stream_handler_invoked = False
+
+    def test_stream_handler(token: str) -> str:
+        # no use of print() here as we are running tests
+        # print(token, flush=True, end="")
+        nonlocal stream_handler_invoked
+        stream_handler_invoked = True
+        return token
+
+    node = PromptNode(
+        "text-davinci-003",
+        api_key=os.environ.get("OPENAI_API_KEY"),
+        model_kwargs={"stream_handler": test_stream_handler},
+    )
+    response = node("What are some of the best cities in the world to live?")
+
+    assert len(response[0]) > 0, "Response should not be empty"
+    assert stream_handler_invoked, "Stream handler should have been invoked"
+
+
```

Comment:
I think these are already end to end tests and should go in the `e2e` folder (like all other tests that do inference). We should still have a couple of unit tests in here, but they should mock `requests` to just make sure the REST call is invoked properly.

Do you think you can take the time for this fix? Otherwise I'd at least open an issue to fix that later.